### PR TITLE
[#335] Remove XML code from Compose template

### PR DESCRIPTION
**Pull request by chornerman**
https://github.com/nimblehq/android-templates/issues/335

## What happened 👀

Remove XML code from Compose template

## Insight 📝

- Remove XML code related dependencies
- Remove `HomeFragment` and its view model, layout, and dimension
- Remove base classes for XML screen and their usages
- Restructure Home Compose screen
- Set the Home Compose screen as a start destination for app navigation

### CODE REVIEW

File: template-compose/app/build.gradle.kts [hunk 1]
```diff
@@ -124,7 +124,6 @@ dependencies {
 
     implementation("androidx.activity:activity-ktx:${Versions.ANDROIDX_ACTIVITY_KTX_VERSION}")
     implementation("androidx.appcompat:appcompat:${Versions.ANDROIDX_SUPPORT_VERSION}")
```
Comment (kaungkhantsoe) [inline template-compose/app/build.gradle.kts:126]: We don't need them anymore
```suggestion
```

Comment (kaungkhantsoe) [inline template-compose/app/build.gradle.kts:126]:
```suggestion
    
```

Comment (kaungkhantsoe) [reply template-compose/app/build.gradle.kts:126]: Shall we also change the extension from `AppCompatActivity` to `ComponentActivity` in `BaseActivity` according to [this thread](https://nimble-co.slack.com/archives/G41UCTGHL/p1669019803168809)? Do we even need `BaseActivity`? @Tuubz

Comment (Tuubz) [reply template-compose/app/build.gradle.kts:126]: I think we should keep "big" changes for separate pull issues (same as switching navigation component tp Kotlin for example).

Would you mind creating an issue for this? 🙏

File: template-compose/app/src/main/java/co/nimblehq/template/compose/ui/screens/home/HomeComposeFragment.kt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package co.nimblehq.template.compose.ui.screens.compose
+package co.nimblehq.template.compose.ui.screens.home
```
Comment (luongvo) [inline template-compose/app/src/main/java/co/nimblehq/template/compose/ui/screens/home/HomeComposeFragment.kt:1]: Can we rename this to HomeFragment now?

File: template-compose/app/src/main/java/co/nimblehq/template/compose/ui/screens/home/HomeComposeViewModel.kt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package co.nimblehq.template.compose.ui.screens.compose
+package co.nimblehq.template.compose.ui.screens.home
```
Comment (luongvo) [inline template-compose/app/src/main/java/co/nimblehq/template/compose/ui/screens/home/HomeComposeViewModel.kt:1]: Can we rename this to HomeViewModel now?

**Comment by Wadeewee** [comment]
@chornerman I think we can remove `Fragment.provideActivityViewModels` and `ComponentActivity.provideViewModels` in  `ViewModelExt.kt` but for `Fragment.provideViewModels` we can remove it later once we remove `fragment` component What do you think? @Tuubz @luongvo

**Comment by luongvo** [comment]
> @chornerman I think we can remove `Fragment.provideActivityViewModels` and `ComponentActivity.provideViewModels` in `ViewModelExt.kt` but for `Fragment.provideViewModels` we can remove it later once we remove `fragment` component What do you think? @Tuubz @luongvo

@Wadeewee Actually, I can remove all of them when working on a `Setup nav graph and navigation logic` task.
Sample PR: https://github.com/nimblehq/jetpack-compose-crypto/pull/49

File: template-compose/app/src/main/java/co/nimblehq/template/compose/ui/theme/Theme.kt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package co.nimblehq.template.compose.ui.screens.compose.theme
```
Comment (Wadeewee) [inline template-compose/app/src/main/java/co/nimblehq/template/compose/ui/theme/Theme.kt:1]: @chornerman To get rid of  `@Suppress("FunctionNaming")`  Let's define `functionPattern` in `detekt-config` 🙏

**Comment by Wadeewee** [comment]
> > @chornerman I think we can remove `Fragment.provideActivityViewModels` and `ComponentActivity.provideViewModels` in `ViewModelExt.kt` but for `Fragment.provideViewModels` we can remove it later once we remove `fragment` component What do you think? @Tuubz @luongvo
> 
> @Wadeewee Actually, I can remove all of them when working on a `Setup nav graph and navigation logic` task. Sample PR: [nimblehq/jetpack-compose-crypto#49](https://github.com/nimblehq/jetpack-compose-crypto/pull/49)

Yeah, @luongvo, we should remove all of them but not for now. That's why I said:

> for `Fragment.provideViewModels`, we can remove it later once we remove `fragment` component 

Because we can't refactor for all of them in this `PR` as it's out of scope and It's a big change 🙏  cc @Tuubz @chornerman

File: template-compose/app/src/main/java/co/nimblehq/template/compose/ui/theme/Theme.kt (hunk 4 above)
Comment (luongvo) [reply template-compose/app/src/main/java/co/nimblehq/template/compose/ui/theme/Theme.kt:1]: We have a separate task for this https://github.com/nimblehq/android-templates/issues/303 cc @Tuubz

Comment (Wadeewee) [reply template-compose/app/src/main/java/co/nimblehq/template/compose/ui/theme/Theme.kt:1]: Hmm, But we already add it in `sample-compose` 🤔

**Comment by chornerman** [comment]
> > > @chornerman I think we can remove `Fragment.provideActivityViewModels` and `ComponentActivity.provideViewModels` in `ViewModelExt.kt` but for `Fragment.provideViewModels` we can remove it later once we remove `fragment` component What do you think? @Tuubz @luongvo
> > 
> > 
> > @Wadeewee Actually, I can remove all of them when working on a `Setup nav graph and navigation logic` task. Sample PR: [nimblehq/jetpack-compose-crypto#49](https://github.com/nimblehq/jetpack-compose-crypto/pull/49)
> 
> Yeah, @luongvo, we should remove all of them but not for now. That's why I said:
> 
> > for `Fragment.provideViewModels`, we can remove it later once we remove `fragment` component
> 
> Because we can't refactor for all of them in this `PR` as it's out of scope and It's a big change 🙏 cc @Tuubz @chornerman
> 
> **Note: I request this because those methods have been unused. Then we can remove them for now.**

@Wadeewee @luongvo I removed unused functions in this PR 🙏🏼

File: template-compose/app/build.gradle.kts (hunk 1 above)
Comment (chornerman) [reply template-compose/app/build.gradle.kts]: @kaungkhantsoe Thank you for spotting out 👍🏼 updated

File: template-compose/app/src/main/java/co/nimblehq/template/compose/ui/theme/Theme.kt (hunk 4 above)
Comment (luongvo) [reply template-compose/app/src/main/java/co/nimblehq/template/compose/ui/theme/Theme.kt:1]: I think we better do re-check all Detekt config for all templates & samples in #303 task @Wadeewee @Tuubz

**Review by luongvo — APPROVED** [review]
GJ

**Review by kaungkhantsoe — APPROVED** [review]

**Review by hoangnguyen92dn — APPROVED** [review]

**Review by Wadeewee — APPROVED** [review]

**Review by Tuubz — COMMENTED** [review]
@chornerman Rest LGTM 🚀

File: template-compose/app/src/main/java/co/nimblehq/template/compose/ui/base/BaseNavigator.kt [hunk 5]
```diff
@@ -76,7 +76,7 @@ abstract class BaseNavigatorImpl(
     }
 
     private fun handleError(error: Throwable) {
-        if (fragment is BaseFragment<*>) {
+        if (fragment is BaseComposeFragment) {
```
[the remark this answers is not in the document]
Comment (Tuubz) [reply template-compose/app/src/main/java/co/nimblehq/template/compose/ui/base/BaseNavigator.kt:79]: @chornerman One last comment left 🚀

**Pull request merged**